internal/observability/graph.go

```diff
@@ -98,15 +112,33 @@ func (g *graph) begin(ctx context.Context, rpcType transport.Type, direction dir
 	now := _timeNow()
 
 	d := digester.New()
-	d.Add(req.Caller)
-	d.Add(req.Service)
-	d.Add(req.Transport)
-	d.Add(string(req.Encoding))
-	d.Add(req.Procedure)
-	d.Add(req.RoutingKey)
-	d.Add(req.RoutingDelegate)
-	d.Add(string(direction))
-	d.Add(rpcType.String())
+	if _, ok := g.metricTagsBlockMap[_source]; !ok {
```

Comment:
@AllenLuUber  and I were chatting offline about it, thinking that map lookup for each key (twice) is an additional overhead. The possible values in block list are known (request struct fields), we could make this one shot cost while constructing graph by mapping block list items into a struct. The block list struct can be used while checking block status inplace of the current map lookups

Reply:
+1. At first I didn't expand and thought this was purely in the ctor (and so I was going to argue that we should just do whatever's most readable/maintainable), but then I saw this is in `begin()` - so yeah, we should use fields instead. Even just something like

```go
type AllowedTags struct {
  Source    bool
  Dest      bool
  Transport bool
  // ...
}

// later...

if g.allowedTags.Source {
  d.add(req.Source)
}

// ...
```

would be a nice marginal improvement.